test/integration/export_test.py

```diff
@@ -0,0 +1,121 @@
+import io
+import csv
+import pytest
+import pyexasol
+from inspect import cleandoc
+
+
+@pytest.fixture
+def connection(dsn, user, password, schema):
+    with pyexasol.connect(
+        dsn=dsn, user=user, password=password, schema=schema, compression=True
+    ) as con:
+        yield con
+
+
+@pytest.fixture
+def table_name():
+    yield "CLIENT_NAMES"
+
+
+@pytest.fixture
+def empty_table(connection, table_name):
+    ddl = cleandoc(f"""
+    CREATE TABLE IF NOT EXISTS {table_name}
+    (
+        FIRSTNAME       VARCHAR(255),
+        LASTNAME        VARCHAR(255)
+    );
+    """)
+    connection.execute(ddl)
+    connection.commit()
+
+    yield table_name
+
+    ddl = f"DROP TABLE IF EXISTS {table_name};"
+    connection.execute(ddl)
+    connection.commit()
+
+
+@pytest.fixture
+def data(faker):
+    yield [(faker.first_name(), faker.last_name()) for _ in range(0, 10)]
+
+
+@pytest.fixture
+def swaped_data(data):
+    yield [(lastname, firstname) for firstname, lastname in data]
+
+
+@pytest.fixture
+def table(connection, empty_table, data):
+    name = empty_table
+    stmt = "INSERT INTO {table} VALUES ({{col1}}, {{col2}});"
+    for col1, col2 in data:
+        connection.execute(stmt.format(table=name), {"col1": col1, "col2": col2})
+    connection.commit()
+    yield name
+
+
+@pytest.fixture
+def export_file(tmp_path, data):
+    yield tmp_path / "names.csv"
+
+
+@pytest.fixture
+def csv_dialect():
+    class PyexasolCsvDialect(csv.Dialect):
+        lineterminator = "\n"
+        delimiter = ","
+        quoting = csv.QUOTE_MINIMAL
+        quotechar = '"'
+
+    yield PyexasolCsvDialect()
+
+
+@pytest.fixture
+def expected_csv(csv_dialect):
+    def create_csv(table, data, **params):
+        csvfile = io.StringIO()
+        if "with_column_names" in params and params["with_column_names"]:
+            data = [("FIRSTNAME", "LASTNAME")] + data
+        writer = csv.writer(csvfile, csv_dialect)
+        writer.writerows(data)
```

Comment:
Does the csv.writer needs to be closed?

Reply:
AFIK no, it takes a file like object as it's first parameter. From my understanding the one providing the file like object is responsible to close the provided file-like object afterwards (later).